app/src/main/java/com/xa/xpensauditor/misc_diff.java

```diff
@@ -0,0 +1,1374 @@
+//             @Override
+//             public void onItemClick(AdapterView<?> parent, View view, int position, long id) {
+//                 String clickedItem = (String) listView.getItemAtPosition(position);
+//                 Toast.makeText(SearchProduct.this, clickedItem, Toast.LENGTH_SHORT).show();
+//             }
+//         });
+
+//        search_button.setOnClickListener(new View.OnClickListener() {
+//             @Override
+//             public void onClick(View view) {
+//                 String text = input.getText().toString();
+//                 if (text == null || text.length() == 0) {
+//                     makeToast("Enter a valid product name.");
+//                 } else {
+//                     new FetchProduct().start();
+//                     addItem();
+//                 }
+//             }
+//         });
+//         loadContent();
+//     }
+
+//     public void loadContent() {
+//         File path = getApplicationContext().getFilesDir();
+//         File readFrom = new File(path, "list.txt");
+//         byte[] content = new byte[(int) readFrom.length()];
+
+//         FileInputStream stream = null;
+//         try {
+//             stream = new FileInputStream(readFrom);
+//             stream.read(content);
+//             String s = new String(content);
+//             s = s.substring(1, s.length() - 1);
+//             String split[] = s.split(", ");
+
+//             if (split.length == 1 && split[0].isEmpty())
+//                 rows = new ArrayList<>();
+//             else rows = new ArrayList<>(Arrays.asList(split));
+
+//             adapter = new ListViewAdapter(this, rows);
+//             listView.setAdapter(adapter);
+//         } catch (Exception e) {
+//             e.printStackTrace();
+//         }
+//     }
+
+//     public static void addItem() {
+//         listView.setAdapter(adapter);
+//     }
+
+//     class FetchProduct extends Thread{
+//         String data = "";
+
+//         @Override
+//         public void run(){
+
+//             fetchHandler.post(new Runnable() {
+//                 @Override
+//                 public void run() {
+//                     progressDialog = new ProgressDialog(SearchProduct.this);
+//                     progressDialog.setMessage("Fetching Data");
+//                     progressDialog.setCancelable(false);
+//                     progressDialog.show();
+//                 }
+//             });
+//             try {
+//                 String api_link = "https://online-product-scrapper.onrender.com/search_walmart?query="+input.getText().toString();
+//                 URL url = new URL(api_link);
+//                 HttpURLConnection httpURLConnection = (HttpURLConnection) url.openConnection();
+//                 InputStream inputStream = httpURLConnection.getInputStream();
+//                 BufferedReader bufferedReader = new BufferedReader(new InputStreamReader(inputStream));
```

Comment:
⚠️ **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.sizes.LineLengthCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Line is longer than 100 characters (found 106).